Docs: How To Use A Capacitor Plugin (#56)
ionic-team/ionic-portals: website/docs/how-to/using-a-capacitor-plugin.md

title: How To Use a Capacitor Plugin
sidebar_label: Use a Capacitor Plugin
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> Ionic Portals uses Capacitor under the hood, meaning that you can use existing Capacitor Plugins in your Portals. These plugins allow Portals to use native functionality without much setup on either the native or web developers part.

Reviewer: might want to say "existing Capacitor Core plugins" with a link to either a doc here or the core plugins page in the cap docs

<Tabs 
    defaultValue="ios" 
    values={[
        { label: 'iOS', value: 'ios', },
        { label: 'Android', value: 'android', },
    ]}
>
<TabItem value="ios">

## Native Usage
In order to use a Capacitor Plugin, you need to install the plugin as a dependency in your `Podfile`.

> ```ruby

Reviewer: change this to ```ruby Podfile {2)``` and take out the podfile comment. The {2} will highlight the line for them to add so it calls it out better.

# Podfile
pod 'IonicPortals', '~> 0.2.0'
pod 'CapacitorStorage', '~> 1.2.0'
```

> :::warning
> Make sure that the versions in your `Podfile`, `build.gradle`, and `package.json` all match! Otherwise you will run into errors.
> :::

Reviewer 2: Suggesting making the Admonition less scary/red and wording this piece for iOS specifically in the iOS tab.

```suggestion
:::caution
Make sure that the versions in your `Podfile` and `package.json` match! Otherwise you could run into errors.
:::
```

**CapacitorNetwork**

The Network API provides network and connectivity information.

**CapacitorPushNotifications**

The Push Notifications API provides access to native push notifications.

**CapacitorScreenReader**

The Screen Reader API provides access to TalkBack/VoiceOver/etc. and provides simple text-to-speech capabilities for visual accessibility.

**CapacitorShare**

The Share API provides methods for sharing content in any sharing-enabled apps the user may have installed.

**CapacitorSplashScreen**

The Splash Screen API provides methods for showing or hiding a Splash image.

**CapacitorStatusBar**

The StatusBar API Provides methods for configuring the style of the Status Bar, along with showing or hiding it.

**CapacitorStorage**

The Storage API provides a simple key/value persistent store for lightweight data.

**CapacitorTextZoom**

The Text Zoom API provides the ability to change Web View text size for visual accessibility.

**CapacitorToast**

The Toast API provides a notification pop up for displaying important information to a user. Just like real toast!

</TabItem>
<TabItem value="android">

> ## Usage 

Reviewer 2: The iOS one says `Native Usage` but the Android one says `Usage` not sure if you intended those to be different

In order to use a Capacitor Plugin, you need to install the plugin as a dependency in your `build.gradle` file.

```groovy
dependencies {
    implementation 'io.ionic:portals:0.2.0'
    implementation 'com.capacitorjs.storage:1.2.0'
}
```

> :::warning
> Make sure that the versions in your `Podfile`, `build.gradle`, and `package.json` all match! Otherwise you will run into errors.
> :::

Reviewer 2: Suggesting making the Admonition less scary/red and wording this piece for Android specifically in the Android tab.

```suggestion
:::caution
Make sure that the versions in your `build.gradle` and `package.json` match! Otherwise you could run into errors.
:::
```

> and include it via the [PortalBuilder.setPlugins()](../reference/android/portal-builder#setplugins) or [PortalBuilder.addPlugin()](../reference/android/portal-builder#addplugin) functions.

Reviewer 2: ```suggestion
Next, include it via the [PortalBuilder.setPlugins()](../reference/android/portal-builder#setplugins) or [PortalBuilder.addPlugin()](../reference/android/portal-builder#addplugin) functions.
```
